gangplank/spec/stages.go

```diff
@@ -524,20 +564,16 @@ func addShorthandToStage(artifact string, stage *Stage) {
 }
 
 // GenerateStages creates stages.
-func (j *JobSpec) GenerateStages(fromNames []string, singleStage bool) {
-	if len(fromNames) == 0 {
-		return
-	}
-
+func (j *JobSpec) GenerateStages(fromNames, testNames []string, singleStage bool) {
 	j.DelayedMetaMerge = true
 	j.Job.StrictMode = true
 
 	if singleStage && len(fromNames) > 0 {
-		newList := []string{strings.Join(fromNames, "+")}
+		newList := []string{strings.Join(append(fromNames, testNames...), "+")}
 		fromNames = newList
 	}
 
-	for _, k := range fromNames {
+	for _, k := range append(fromNames, testNames...) {
```

Comment:
Not a big issue but users will be able to pass test names as build artifacts i.e. `gangplank generate -A basicBios`.  It will still create test stage correctly.

Reply:
Got catch. Last push has a validation for that:
```
ben@porpentina:~/src/cosa/gangplank $ bin/gangplank podless --kolaTest aws
INFO[0000] Gangplank: COSA OpenShift job runner, 2021-04-29.f8577630~dirty 
INFO[0000] Initalizing Build Tree                        dir=
INFO[0000] Generating jobspec from CLI arguments        
INFO[0000] Generating stages                            
FATA[0000] failed to generate the jobpsec                error="kola test aws is an invalid kola name"

ben@porpentina:~/src/cosa/gangplank $ bin/gangplank podless -A basicBios
INFO[0000] Gangplank: COSA OpenShift job runner, 2021-04-29.f8577630~dirty 
INFO[0000] Initalizing Build Tree                        dir=
INFO[0000] Generating jobspec from CLI arguments        
INFO[0000] Generating stages                            
FATA[0000] failed to generate the jobpsec                error="artifact basicBios is an invalid artifact"
```